src/Toml/Parser/Value.hs

```diff
@@ -145,6 +146,10 @@ tableNameP :: Parser Key
 tableNameP = between (text "[") (text "]") keyP
 
 
+tableArrayNameP :: Parser Key
+tableArrayNameP = lexeme $ between (text "[[") (text "]]") keyP
```

Comment:
Actually, I should instead remove the `lexeme` here, because the `text "]]"` already includes a `lexeme`.
`Toml.Parser.Core.parse (tableHeaderP >> eof) "" "[a]   "` returns `Right ()`.